google/auth/credentials_async.py

```diff
@@ -0,0 +1,168 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+"""Interfaces for credentials."""
+
+import abc
+
+import six
```

Comment:
Should six be needed for this? this surface is only usable for 3.6+ correct?

Reply:
I used six to change the exception types (line 143, 147) so that we could raise our defined transport error.

Reply:
This module uses `six` only for the `six.add_metaclass(abc.ABCMeta)` class decorator.  Given that it will only ever be imported under Python >= 3.6, it would be clearer to just use the direct Py3 syntax, e.g.:

```python
class Credentials(credentials.Credentials, metaclass=abc.ABCMeta):
    ...
```